recipes/jsoncons/all/conanfile.py

```diff
@@ -0,0 +1,43 @@
+from conan import ConanFile
+from conan.tools.files import get, copy
+from conan.tools.layout import basic_layout
+import os
+
+required_conan_version = ">=1.50.0"
+
+
+class JsonconsConan(ConanFile):
+    name = "jsoncons"
+    description = "A C++, header-only library for constructing JSON and JSON-like data formats, with JSON Pointer, JSON Patch, JSON Schema, JSONPath, JMESPath, CSV, MessagePack, CBOR, BSON, UBJSON"
+    topics = ("json", "csv", "cpp", "json-serialization", "cbor", "json-parser", "messagepack", "json-pointer", "json-patch", "json-diff", "bson", "ubjson", "json-parsing", "jsonpath", "jmespath", "csv-parser", "csv-reader", "jsonschema", "json-construction", "streaming-json-read")
+    homepage = "https://github.com/danielaparker/jsoncons"
+    url = "https://github.com/danielaparker/jsoncons"
+    author = "Daniel Parker danielaparker@yahoo.com"
+    license = "Boost Software License"
+    no_copy_source = True
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def source(self):
+        get(
+            self,
+            url=f"https://github.com/danielaparker/jsoncons/archive/refs/tags/v{self.version}.tar.gz", 
```

Comment:
Please see the templates for how this should be implemented https://github.com/conan-io/conan-center-index/blob/94dddc7d36d58fa87ea327b7cdff44787da7db93/docs/package_templates/header_only/all/conanfile.py#L78